Userland/Libraries/LibHTTP/Job.cpp

```diff
@@ -382,9 +391,9 @@ void Job::finish_up()
         // before we can actually call `did_finish`. in a normal flow, this should
         // never be hit since the client is reading as we are writing, unless there
         // are too many concurrent downloads going on.
-        deferred_invoke([this](auto&) {
-            finish_up();
-        });
+        dbgln_if(JOB_DEBUG, "Flush finished with {} bytes remaining, will try again later", m_buffered_size);
+        if (!has_timer())
+            start_timer(50);
```

Comment:
This is an arbitrary guess, not sure if it should be larger or smaller...I feel like 50ms is enough time for the client to get back to its event loop from whatever it's doing